### PR TITLE
nerf deadened pain awareness

### PR DESCRIPTION
**Pull request by KorGgenT**
#### Summary

```SUMMARY: Balance "Nerf Deadened pain awareness"```
#### Purpose of change

If you can't feel any pain, you have to pay very close attention to your extremities, and even then you won't really be able to tell for sure how good or bad they are without very close examination and even some medical know-how. As such, the nurse bot can still scan you to give you the self-aware-like effect.
#### Describe the solution

Check for NOPAIN and alter the display.

unknown

### CODE REVIEW

**Comment by DemAvalon** [comment]
Deadened mutants are still capable of felling the description is 

"Nothing hurts any more. Those bites tickle and the wounds itch a bit, but that's it."
--
 
So they would still be aware of damage being done and the severity based on how "ticklish" or "itchy" the wound feels.

**Comment by Amoebka** [comment]
This implies regular characters estimate the severity of their wounds by how much they hurt (which isn't a mechanic in the game) and that pain is a more precise indicator than visual examination (which is just plain stupid).

**Comment by KorGgenT** [comment]
> This implies regular characters estimate the severity of their wounds by how much they hurt (which isn't a mechanic in the game) and that pain is a more precise indicator than visual examination (which is just plain stupid).

all right, then maybe we should move away from the bars to something a little less exact. maybe the bars can be the "self aware"

**Comment by DemAvalon** [comment]
> all right, then maybe we should move away from the bars to something a little less exact. maybe the bars can be the "self aware"

Is a video game, being vague about how many hitpoints the player has sounds like an awful idea.

**Comment by Amoebka** [comment]
I mean, if you feel like rebalancing the whole system, the accuracy of wound estimations should probably depend on first aid skill instead of being a binary self-aware/non-self-aware.

I personally would't do that, though. If the player can't even tell how much HP they have beyond "good" and "bad", how are they even supposed to make informed choices and play the game?

**Comment by I-am-Erk** [comment]
How do you figure players know how their limbs are doing currently if not through sensory input?

This is a pretty accurate depiction of how sensory neuropathies behave in real life.
I would suggest that there be a way to more closely examine your wounds, eg. getting a more detailed readout in the @ menu

**Comment by kevingranade** [comment]
> It's a video game, being vague about how many hitpoints the player has sounds like an awful idea.

That doesn't follow at all, what the player is and is not aware of is absolutely part of the game, less precise feedback means the player needs to make decisions in a different way, but it doesn't imply it's per se bad.

> I mean, if you feel like rebalancing the whole system, the accuracy of wound estimations should probably depend on first aid skill instead of being a binary self-aware/non-self-aware.

> I would suggest that there be a way to more closely examine your wounds, eg. getting a more detailed readout in the @ menu

This seems reasonable, you get vague feedback about how much it hurts (and deadened or painkillers reduce this feedback), and a manual inventory via a self-examination activity can give you more precise information.

**Comment by DemAvalon** [comment]
> This is a pretty accurate depiction of how sensory neuropathies behave in real life.

Except the deadened mutant is not completely numb and as I said would be able to gauge how wounded they are based on how "ticklish" or "itchy" their limbs feel.

**Comment by I-am-Erk** [comment]
Sensory neuropathies do not all leave people feeling completely numb, yet people continue to get severe wounds when they have dramatically reduced sensation.

To put it differently, if you're *almost* deaf and can still hear a tiny bit, that doesn't mean that you must be able to follow a spoken conversation. Or if you're incredibly nearsighted, you're not blind but that doesn't mean you can read a book without glasses.

**Comment by DemAvalon** [comment]
> This seems reasonable, you get vague feedback about how much it hurts (and deadened or painkillers reduce this feedback), and a manual inventory via a self-examination activity can give you more precise information.

This sounds like an unnecessary hassle, it would add extra seconds for the player to be able to how they actually are without any gain.

Obfuscating crucial information to the player (how much hp they have) sounds very user-hostile.

**Comment by DemAvalon** [comment]
I would like to apologize, after reading my comments later, I realized I might be coming off as rude, that was not my intention, I was just kind of salty I guess (I don't even know why I got upset I don't go for medical) so, I'm sorry, I still don't like the changes but I understand the rationale behind it.

File: src/panels.cpp [hunk 1]
```diff
@@ -882,22 +884,31 @@ static void draw_limb_health( avatar &u, const catacurses::window &w, int limb_i
...
-    const auto &hp = get_hp_bar( u.hp_cur[limb_index], u.hp_max[limb_index] );
+    std::pair<std::string, nc_color> hp = get_hp_bar( u.hp_cur[limb_index], u.hp_max[limb_index] );
 
     if( is_self_aware || u.has_effect( effect_got_checked ) ) {
         wprintz( w, hp.second, "%3d  ", u.hp_cur[limb_index] );
+    } else if( no_feeling ) {
+        if( u.hp_cur[limb_index] < u.hp_max[limb_index] / 2 ) {
+            hp = std::make_pair( string_format( " %s", _( "Bad" ) ), c_red );
+        } else {
+            hp = std::make_pair( string_format(  " %s", _( "Good" ) ), c_green );
```
Comment (kevingranade) [inline src/panels.cpp]:
```suggestion
            hp = std::make_pair( string_format( " %s", _( "Good" ) ), c_green );
```

**Comment by nexusmrsep** [comment]
There is a body parts menu used when you use bandages, disinfectants etc that already displays some extra medical info. Perhaps it can be invoked as an action as a way to do detailed examination of one's condition.

**Comment by Fris0uman** [comment]
Won't having to examibe yourself to get your hp add lot of tedium ?
Whatever check is put into that action could be applied for the Hp displayed on the side panel without requiring more player input. Unless we want it to take in-game time

**Comment by kevingranade** [comment]
Yes I was suggesting that the self examination action would take time, most likely enough that it would be impractical to perform in combat.

**Pull request merged**